shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -244,14 +315,124 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
+            }
+          }
+
+          def convertColumnarBatchToInternalRowIter(): Iterator[InternalRow] = {
+            val parquetCachedBatch = cbIter.next().asInstanceOf[ParquetCachedBatch]
+            val inputFile = new ByteArrayInputFile(parquetCachedBatch.buffer)
+
+            val catalystColumns =
+              getColumnIndices(cacheAttributes).map { i => "_col" + i }
+
+            val catalystSchema = cacheAttributes.zipWithIndex.map {
+              case (attr, i) => attr.withName(catalystColumns(i))
+            }
+
+            val selectedColumns =
+              getColumnIndices(selectedAttributes).map { i => "_col" + i }
+
+            val requestedSchema = selectedAttributes.zipWithIndex.map {
+              case (attr, i) => attr.withName(selectedColumns(i))
+            }
+
+            val sharedConf = broadcastedHadoopConf.value.value
+            val options = HadoopReadOptions.builder(sharedConf).build()
+            val parquetFileReader = ParquetFileReader.open(inputFile, options)
+            val footerFileMetaData = parquetFileReader.getFileMetaData
+            val parquetSchema = parquetFileReader.getFooter.getFileMetaData.getSchema
+
+            val convertTz = None // as we are using hadoop and parquet mr
+
+            val unsafeRows = new ArrayBuffer[InternalRow]
+            import org.apache.parquet.io.ColumnIOFactory
+            var pages = parquetFileReader.readNextRowGroup()
+            while (pages != null) {
+              val rows = pages.getRowCount
+              val columnIO = new ColumnIOFactory().getColumnIO(parquetSchema)
+              val recordReader =
+                columnIO.getRecordReader(pages, new ParquetRecordMaterializer(parquetSchema,
+                  getSchemaFromAttributeSeq(catalystSchema),
+                  new ParquetToSparkSchemaConverter(sharedConf), convertTz,
+                  LegacyBehaviorPolicy.CORRECTED))
+              for (i <- 0 until rows.toInt) {
+                val row = recordReader.read
+                unsafeRows += row.copy()
+              }
+              pages = parquetFileReader.readNextRowGroup()
+            }
+            parquetFileReader.close()
+            val iter = unsafeRows.iterator
+            val unsafeProjection =
+              GenerateUnsafeProjection.generate(requestedSchema, requestedSchema)
+            iter.map(unsafeProjection)
+          }
+        }
+      }
+    }
+  }
+
+  private def getHadoopConf(requestedSchema: StructType,
+     sqlConf: SQLConf) : Configuration = {
+    val hadoopConf = new Configuration()
+
+    hadoopConf.set(ParquetInputFormat.READ_SUPPORT_CLASS, classOf[ParquetReadSupport].getName)
+    hadoopConf.set(
+      ParquetReadSupport.SPARK_ROW_REQUESTED_SCHEMA,
+      requestedSchema.json)
+    hadoopConf.set(
+      ParquetWriteSupport.SPARK_ROW_SCHEMA,
+      requestedSchema.json)
+    // we support UTC time only on cudf
+    hadoopConf.set(
+      SQLConf.SESSION_LOCAL_TIMEZONE.key,
+      "UTC")
```

Comment:
What if the JVM timezone (and thus Spark timestamps) aren't UTC?  Is it safe to make this override?

Reply:
Thats a good catch. We need this to reflect the system timezone as we are using this config to write on CPU and not cudf